neptun/src/noise/session.rs

```diff
@@ -194,14 +193,18 @@ impl Session {
     /// src - an IP packet from the interface
     /// dst - pre-allocated space to hold the encapsulating UDP packet to send over the network
     /// returns the size of the formatted packet
-    pub(super) fn format_packet_data<'a>(&self, src: &[u8], dst: &'a mut [u8]) -> &'a mut [u8] {
-        if dst.len() < src.len() + super::DATA_OVERHEAD_SZ {
+    pub(super) fn format_packet_data<'a>(
+        &self,
+        buffer: &'a mut [u8],
+        data_len: usize,
+    ) -> &'a mut [u8] {
+        if buffer.len() < data_len + super::DATA_OVERHEAD_SZ {
             panic!("The destination buffer is too small");
```

Comment:
I've experienced reproducible panic in this place when running iperf3 -u -b 500M -i 60 -t 120 -c 10.0.2.2 locally

Reply:
Looking into it.